packages/edit-site/src/components/block-editor/back-button.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * WordPress dependencies
+ */
+import { Button } from '@wordpress/components';
+import { arrowLeft } from '@wordpress/icons';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { store as editSiteStore } from '../../store';
+import useTemplateTitle from '../use-template-title';
+
+function getTemplateTypeName( templateType ) {
+	switch ( templateType ) {
+		case 'wp_template_part':
+			return __( 'template part' );
+		case 'wp_template':
+			return __( 'template' );
+		default:
+			throw new Error( 'Unknown template type' );
+	}
+}
+
+function BackButton() {
+	const {
+		isTemplatePart,
+		previousTemplateType,
+		previousTemplateId,
+	} = useSelect( ( select ) => {
+		const {
+			getEditedPostType,
+			getPreviousEditedPostType,
+			getPreviousEditedPostId,
+		} = select( editSiteStore );
+
+		return {
+			isTemplatePart: getEditedPostType() === 'wp_template_part',
+			previousTemplateType: getPreviousEditedPostType(),
+			previousTemplateId: getPreviousEditedPostId(),
+		};
+	}, [] );
+	const previousTemplateTitle = useTemplateTitle(
+		previousTemplateType,
+		previousTemplateId
+	);
+	const { goBack } = useDispatch( editSiteStore );
+
+	if ( ! isTemplatePart || ! previousTemplateId ) {
+		return null;
+	}
+
+	return (
+		<Button
```

Comment:
We probably should auto focus on this element when navigating to the isolated editor?

Reply:
Hm. Could we focus `.edit-site-visual-editor`? That way pressing <kbd>Tab</kbd> once focuses the close button. This matches how other modal interfaces work in the block editor e.g. opening Preferences.

cc. @tellthemachines who is smarter than I am at this kind of stuff.

Reply:
I'm not 100% sure what's going on here, but this is what I'm seeing:

When tabbing through the page, starting from the header, the back button is focusable in the normal tab order _as long as nothing inside the editor is selected_. This means that if I tab to the button, and tab once more, landing on the first block in the editor area, I'm unable to Shift+Tab back to the button. In addition, the shortcut to navigate between landmark areas (Ctrl+`) doesn't seem to work in the site editor, so there's no way to get back to that button.

I'd expect to be able to Shift+Tab back to the button from the editor.

Reply:
Hmm, I couldn't figure it out either. It seems like pressing <kbd>Shift</kbd> + <kbd>Tab</kbd> goes back to the editor header in site editor. No idea if it's a browser thing or it's handled somewhere else 🤔. It's not that normal since the editor is in an iframe. Anyone has any idea how to fix this?